lib/adapters/tool/playwright/index.ts

```diff
@@ -0,0 +1,311 @@
+    private _haveProjectsInConfig: boolean;
+    private _htmlReporter: HtmlReporter;
+    private _pwtBinaryPath: string;
+    private _reportBuilder!: GuiReportBuilder;
+    private _eventSource!: EventSource;
+    private _guiApi?: GuiApi;
+
+    static create(
+        this: new (options: ToolAdapterOptionsFromCli) => PlaywrightToolAdapter,
+        options: ToolAdapterOptionsFromCli
+    ): PlaywrightToolAdapter {
+        return new this(options);
+    }
+
+    constructor(opts: ToolAdapterOptionsFromCli) {
+        const {config, configPath} = readPwtConfig(opts);
+
+        this._configPath = configPath;
+        this._config = PlaywrightConfigAdapter.create(config);
+        this._toolName = opts.toolName;
+
+        const pluginOpts = getPluginOptions(this._config.original);
+        this._reporterConfig = parseConfig(pluginOpts);
+
+        this._haveProjectsInConfig = !_.isEmpty(this._config.original.projects);
+
+        this._htmlReporter = HtmlReporter.create(this._reporterConfig, {toolName: ToolName.Playwright});
+        this._pwtBinaryPath = npmWhich.sync(ToolName.Playwright, {cwd: process.cwd()});
+    }
+
+    get configPath(): string {
+        return this._configPath;
+    }
+
+    get toolName(): ToolName {
+        return this._toolName;
+    }
+
+    get config(): PlaywrightConfigAdapter {
+        return this._config;
+    }
+
+    get reporterConfig(): ReporterConfig {
+        return this._reporterConfig;
+    }
+
+    get htmlReporter(): HtmlReporter {
+        return this._htmlReporter;
+    }
+
+    get guiApi(): GuiApi | undefined {
+        return this._guiApi;
+    }
+
+    initGuiApi(): void {
+        this._guiApi = GuiApi.create();
+    }
+
+    async readTests(): Promise<PlaywrightTestCollectionAdapter> {
+        const stdout = await new Promise<string>((resolve, reject) => {
+            // specify default browser in order to get correct stdout with browser name
+            const browserArgs = this._haveProjectsInConfig ? [] : ['--browser', DEFAULT_BROWSER_ID];
+
+            const child = spawn(this._pwtBinaryPath, ['test', '--list', '--reporter', 'list', '--config', this._configPath, ...browserArgs]);
+            let stdout = '';
+            let stderr = '';
+
+            child.stdout.on('data', (data) => stdout += data);
+            child.stderr.on('data', (data) => stderr += data);
+
+            child.on('error', (error) => reject(error));
+
+            child.on('exit', (code) => {
+                if (code !== 0) {
+                    return reject(new Error(`Playwright process with reading tests exited with code: ${code}, stderr: ${stderr}`));
+                }
+
+                resolve(stdout);
+            });
+        });
+
+        const stdoutByLine = stdout.split('\n').map(v => v.trim());
+        const startIndex = stdoutByLine.findIndex(v => v === 'Listing tests:');
+        const endIndex = stdoutByLine.findIndex(v => /total: \d+ tests? in \d+ file/i.test(v));
```

Comment:
Stout of this command looks like this:
```
Listing tests:
  example.spec.js:4:5 › suite #js › has title #1
  example.spec.js:16:5 › suite #js › has title #2
Total: 2 tests in 1 file
```

So I need to get all info between `Listing tests` and `Total: ...`

Reply:
I am 100% sure you've checked this, but do they really lack json-reporter so that we didn't need to parse it like this?